Kuadrant/limitador-operator
# Using limitador cli

### what

* Using limitador cli instead of env vars
* Some cleanup in the makefile
  * remove unused targets
  * `make local-setup` deploys the operator in devel mode

Fixes #46 

### verification steps

dev setup

```
make local-setup
```
deploy limitador
```yaml
k apply -f - <<EOF
---
apiVersion: limitador.kuadrant.io/v1alpha1
kind: Limitador
metadata:
  name: limitador-sample
spec:
  listener:
    http:
      port: 8080
    grpc:
      port: 8081
  limits:
    - conditions: ["get_toy == 'yes'"]
      max_value: 2
      namespace: toystore-app
      seconds: 30
      variables: []
EOF
```
Check limitador process is being configured through the command line parameters, specifically `container[].command` field of the containers
```yaml
k get deployments limitador-sample -o jsonpath='{.spec.template.spec.containers}' | yq_pretty 
- command:
    - limitador-server
    - /home/limitador/etc/limitador-config.yaml
  image: quay.io/kuadrant/limitador:latest
  ...
```

Check deployment is available

```
kubectl wait --timeout=300s --for=condition=Available deployments limitador-sample
deployment.apps/limitador-sample condition met
```
